homeassistant/components/broadlink/switch.py

```diff
@@ -159,25 +157,27 @@ def update_data(self):
 
     async def async_added_to_hass(self):
         """Call when the switch is added to hass."""
-        if self._state is None:
+        if self.state is None:
             state = await self.async_get_last_state()
-            self._state = state is not None and state.state == STATE_ON
+            self._attr_state = state is not None and state.state == STATE_ON
         self.async_on_remove(self._coordinator.async_add_listener(self.update_data))
 
     async def async_update(self):
         """Update the switch."""
         await self._coordinator.async_request_refresh()
+        state = await self.async_get_last_state()
+        self._attr_state = state is not None and state.state == STATE_ON
```

Comment:
I don't think we need this. We are listening to the update coordinator. Whenever there is an update available `update_data` will be called and the state updated.